### PR TITLE
Dev tc state issue 254

### LINKED ISSUE #254: Get TC configuration
Currently, there is no way of getting some of the TC configuration such as conversion mode. There is a way to get the cold junction disabled mode and probe type, however.

I need to validate that I am changing the conversion mode from single to auto/continuous or vice-versa. Because when a module is disabled on data collector, I am supposed to set it to single mode, and when it is enabled, I am suppose to set it to auto/continuous.

### PR DESCRIPTION
**Pull request by sjpark608**
Closes #254 

- get_state for Thermocouple
- TCState class added to update state of the TC configurations
- get_state is called at the end of config() method, so the state is updated after a config is changed

### CODE REVIEW

File: src/edgepi/tc/edgepi_tc.py [hunk 1]
```diff
@@ -254,7 +323,7 @@ def __get_tc_type(self):
         cr1 = Bits(uint=self.__read_register(TCAddresses.CR1_R.value)[1], length=8)
         tc_bits = cr1[-4:].uint
         for enum in TCType:
-            if enum.value.op_code == tc_bits:
+            if not isinstance(enum.value, int) and enum.value.op_code == tc_bits:
                 return enum
```
Comment (jarell-santella) [inline src/edgepi/tc/edgepi_tc.py:327]: Not a big deal but `enum` is also the name of the library you imported to get the `Enum` class. I'm not sure if it's good naming practice to use `enum` again here.

File: src/edgepi/tc/tc_constants.py [hunk 2]
```diff
@@ -276,7 +276,7 @@ class TCType(Enum):
     TYPE_R = OpCode(0x05, TCAddresses.CR1_W.value, Masks.CR1_LOW_MASK.value)  # type R thermocouple
     TYPE_S = OpCode(0x06, TCAddresses.CR1_W.value, Masks.CR1_LOW_MASK.value)  # type S thermocouple
     TYPE_T = OpCode(0x07, TCAddresses.CR1_W.value, Masks.CR1_LOW_MASK.value)  # type T thermocouple
-
+    TYPE_MASK = 0x07
```
Comment (jarell-santella) [inline src/edgepi/tc/tc_constants.py:279]: Why is this pair different from the rest? Is inconsistency ok in this case?

File: src/edgepi/tc/tc_constants.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ class AvgMode(Enum):
     AVG_4 = OpCode(0x20, TCAddresses.CR1_W.value, Masks.CR1_HIGH_MASK.value)  # 4 samples averaged
     AVG_8 = OpCode(0x30, TCAddresses.CR1_W.value, Masks.CR1_HIGH_MASK.value)  # 8 samples averaged
     AVG_16 = OpCode(0x40, TCAddresses.CR1_W.value, Masks.CR1_HIGH_MASK.value)  # 16 samples averaged
-
+    AVG_MASK = 0x70
```
Comment (jarell-santella) [inline src/edgepi/tc/tc_constants.py:265]: Why is this pair different from the rest? Is inconsistency ok in this case?

File: src/edgepi/tc/edgepi_tc.py (hunk 1 above)
Comment (sjpark608) [reply src/edgepi/tc/edgepi_tc.py]: good point and fixed aef1556b20336d5d0c033db5c414ed15cc7184e0

File: src/edgepi/tc/tc_constants.py (hunk 2 above)
Comment (sjpark608) [reply src/edgepi/tc/tc_constants.py]: good point and fixed aef1556b20336d5d0c033db5c414ed15cc7184e0

File: src/edgepi/tc/tc_constants.py (hunk 3 above)
Comment (sjpark608) [reply src/edgepi/tc/tc_constants.py:267]: good point and fixed aef1556b20336d5d0c033db5c414ed15cc7184e0

**Review by naomiven — APPROVED** [review]

**Pull request merged**